test/test_mixed_null_assertions.cpp

```diff
@@ -78,3 +78,126 @@ TEST(List_Mixed_do_insert)
     list.insert_null(0);
     list.insert(0, Mixed("hello world"));
 }
+
+TEST(Mixed_List_unresolved_as_null)
+{
+    Group g;
+    auto t = g.add_table("foo");
+    t->add_column_list(type_Mixed, "mixeds");
+    auto obj = t->create_object();
+    auto obj1 = t->create_object();
+
+    auto list = obj.get_list<Mixed>("mixeds");
+
+    list.insert_null(0);
+    list.insert(1, Mixed{"test"});
+    obj1.invalidate();
+    list.insert(2, obj1);
+
+    CHECK(list.size() == 3);
+
+    {
+        // find all mixed nulls or unresolved link should work the same way
+        list.find_all(realm::null(), [this](size_t pos) {
+            CHECK(pos == 0 || pos == 2);
```

Comment:
I'm pretty sure that this test is not finding 2. Can you make the test stronger so that both results are checked for?

Reply:
I does find pos == 2
![Screenshot 2022-04-04 at 13 04 35](https://user-images.githubusercontent.com/1497069/161540061-af592e06-433b-403f-b089-df2b68fe0b66.png)

Reply:
No, the test was not finding unresolved values. That is because the list in the test did not contain any unresolved links. I have updated your test to include an unresolved link at position 2, and to show you that find_all is failing. If you invalidate an object while it doesn't have any incoming links, it becomes null, while if you invalidate an object when it does have incoming links it becomes unresolved.

Reply:
OK, it should be fixed right now, with the caveat that find_all for a null mixed will result into a linear scan of the list itself in order to find all the unresolved links.